server/service/src/document/patient/query.rs

```diff
@@ -0,0 +1,142 @@
+use chrono::NaiveDate;
+use repository::{
+    DateFilter, EqualFilter, Gender, NameFilter, NameRepository, NameRow, NameSort, NameSortField,
+    NameType, PaginationOption, RepositoryError, SimpleStringFilter, Sort,
+};
+
+use crate::{get_default_pagination_unlimited, service_provider::ServiceContext};
+
+#[derive(Clone, Default)]
+pub struct PatientFilter {
+    pub id: Option<EqualFilter<String>>,
+    pub first_name: Option<SimpleStringFilter>,
+    pub last_name: Option<SimpleStringFilter>,
+    pub gender: Option<EqualFilter<Gender>>,
+    pub date_of_birth: Option<DateFilter>,
+    pub phone: Option<SimpleStringFilter>,
+    pub address1: Option<SimpleStringFilter>,
+    pub address2: Option<SimpleStringFilter>,
+    pub country: Option<SimpleStringFilter>,
+    pub email: Option<SimpleStringFilter>,
+}
+
+#[derive(PartialEq, Debug)]
+pub enum PatientSortField {
+    Name,
+    FirstName,
+    LastName,
+    Gender,
+    DateOfBirth,
+    Phone,
+    Address1,
+    Address2,
+    Country,
+    Email,
+}
+
+pub type PatientSort = Sort<PatientSortField>;
+
+pub struct Patient {
+    pub name_row: NameRow,
+}
+
+impl Patient {
+    pub fn id(&self) -> &String {
```

Comment:
names are not unique

Reply:
I should have a made a comment one line lower, `&self.name_row.name` this is not unique, and can't be treated as ID